src/ghost_borrow_mut.rs

```diff
@@ -248,20 +261,22 @@ generate_public_instance!(a, b, c, d, e, f, g, h, i, j, k, l ; T0, T1, T2, T3, T
 //  Implementation
 //
 
-/// Returns `Err(GhostAliasingError)` if the inputs are distinct, and `Ok(())` otherwise.
-fn check_distinct<const N: usize>(mut arr: [*const (); N]) -> Result<(), GhostAliasingError> {
+/// Returns `Ok(())` if the inputs are distinct, and `Err(GhostAliasingError)` otherwise.
+/// 
+/// Ignores the metadata of the given pointers if they are fat, only comparing their addresses.
```

Comment:
In C++ it is possible to craft a hierarchy like:

    struct B { virtual ~B() = default; virtual void set(int a) = 0; };

    struct D: B, B { void set(int a) override { x = a}; int x; };

In which case two pointers to `B` can be distinct while a call to their `set` methods would involve the same piece of memory.

I don't think that Rust has the same issue _for the moment_. However it will be important to keep an eye on RFCs that wish to extend the trait systems, for example to changes of how slices are modeled as traits.

I would add a further `# Safety` comment there, mentioning that this function's safety relies on the fact that at the moment the data pointer of trait objects always point to the first byte of the concrete object type. This makes the assumption explicit, which may help later on.